### DISCUSSION TITLE
Groups like in zarr?

### DISCUSSION THREAD

**Discussion by Karol-G** [Blosc/python-blosc2#184] [Q&A] [answered]
Hey,

Is it possible to create groups like in Zarr to store arrays of different shapes in the same object (and file)?

Zarr example:

```python
import zarr
g1 = zarr.group()
d1 = g1.create_dataset('foo', shape=(10000, 10000),
                       chunks=(1000, 1000))
d2 = g1.create_dataset('bar/baz/qux', shape=(100, 100, 100),
                       chunks=(100, 10, 10))
```

Is something like this possible in blosc2?

Best regards,
Karol

**Comment by FrancescAlted** [comment, marked as answer]
No, Blosc2 only can store a single dataset (+ metadata) per file, but you can always create your own hierarchy by using the filesystem.  For example, we are currently working on [Caterva2](https://github.com/Blosc/Caterva2), an on demand access to local/remote Blosc2 data repositories, that is able to read an on-disk hierarchical structure of datasets.  Caterva2 also supports plugins for different formats besides Blosc2 (currently HDF5) and visualizers (see e.g. a [3D tomography here](https://demo.caterva2.net/roots/example/lung-jpeg2000_10x.b2nd?roots=example&roots=h5numbers_j2k)).

**Comment by Karol-G** [comment]
Thank you for the quick reply! In my case, groups as in Zarr would be preferable, but I might check out Caterva2.
